docs/proposals/drafts/20211007-pluggable-authenticators.md

```diff
@@ -0,0 +1,146 @@
+---
+toc: true
+title: 2021-10 Pluggable Authenticators
+---
+
+## Introduce Pluggable Authenticators
+
+* **Owners**
+  * @tareqmamari
+
+## TL; DR
+
+We propose a pluggable authenticator approach to support plugging in custom authenticators in a self-registering manner.
+
+## Why
+
+This proposed approach allows the adopters of Observatorium to use their own custom authenticators' implementation other than the current OIDC, mTLS or Openshift authenticators.
+
+## Goals
+
+- Provide a common interface for the authenticators, so that the adopters can use it to implement their own.
+- Provide a self-registering mechanism of the authenticators so that the adopters do not need to patch the current codebase to introduce their own authenticators.
```

Comment:
```suggestion
- Provide a self-registering mechanism for authenticators so that adopters do not need to patch the current codebase to introduce their own authenticators.
```